### PR TITLE
Add disabled_hooks CLI argument

### PR DESCRIPTION
**Pull request by fmeum**
The new argument can be used to selectively disable both custom and and
built-in hooks.

### CODE REVIEW

File: agent/src/main/java/com/code_intelligence/jazzer/agent/Agent.kt [hunk 1]
```diff
@@ -103,7 +104,9 @@ fun premain(agentArgs: String?, instrumentation: Instrumentation) {
         ManifestUtils.combineManifestValues(ManifestUtils.HOOK_CLASSES).flatMap {
             it.split(':')
         }.filter { it.isNotBlank() }
-    val customHookNames = manifestCustomHookNames + (argumentMap["custom_hooks"] ?: emptyList())
+    val customHookNames = (manifestCustomHookNames + (argumentMap["custom_hooks"] ?: emptyList())).filter {
+        it !in (argumentMap["disabled_hooks"]?.toSet() ?: emptySet())
```
Comment (bertschneider) [inline agent/src/main/java/com/code_intelligence/jazzer/agent/Agent.kt:108]: An info log could be nice to detect incidentally disabled hooks.

**Review by bertschneider — COMMENTED** [review]
Looks good, besides the failing tests :grin:

File: tests/BUILD.bazel [hunk 2]
```diff
@@ -162,3 +162,16 @@ JAZZER_API_TEST_CASES = {
...
 ]
+
+java_fuzz_target_test(
+    name = "DisabledHooksFuzzer",
+    timeout = "short",
+    srcs = ["src/test/java/com/example/DisabledHooksFuzzer.java"],
+    expect_crash = False,
+    fuzzer_args = [
+        "-runs=0",
+        "--disabled_hooks=com.example.DisabledHook:com.code_intelligence.jazzer.sanitizers.RegexInjection",
```
[the remark this answers is not in the document]
Comment (fmeum) [reply tests/BUILD.bazel]: Should be fixed, I also updated the help strings.

File: agent/src/main/java/com/code_intelligence/jazzer/agent/Agent.kt (hunk 1 above)
Comment (fmeum) [reply agent/src/main/java/com/code_intelligence/jazzer/agent/Agent.kt]: Good point, added.

**Review by bertschneider — APPROVED** [review]

**Pull request merged**